server/sonar-db-migration/src/test/java/org/sonar/server/platform/db/migration/version/v64/UpgradeQualityTemplateLoadedTemplatesTest.java

```diff
@@ -0,0 +1,172 @@
+    setupDefaultOrganization();
+    insertLoadedTemplate("foo", "bar");
+
+    underTest.execute();
+
+    assertThat(loadedTemplateExists("foo", "bar")).isTrue();
+    assertThat(db.countRowsOfTable(TABLE_LOADED_TEMPLATES)).isEqualTo(1);
+  }
+
+  @Test
+  public void execute_updates_any_row_with_type_QUALITY_PROFILE_to_type_based_on_current_key_md5_and_default_organization_uuid_as_key() throws SQLException {
+    setupDefaultOrganization();
+
+    // put accents to ensure UTF-8 byte array of String is used
+    String key1 = "fé@è:bar";
+    String key2 = "bar";
+    insertLoadedTemplate(QUALITY_PROFILE_TYPE, key1);
+    // matching on type is case sensitive
+    insertLoadedTemplate(QUALITY_PROFILE_TYPE.toLowerCase(), key1);
+    insertLoadedTemplate(QUALITY_PROFILE_TYPE, key2);
+    insertLoadedTemplate("other type", key2);
+
+    underTest.execute();
+
+    assertThat(loadedTemplateExists(QUALITY_PROFILE_TYPE + '.' + md5Hex(key1.getBytes(UTF_8)), DEFAULT_ORGANIZATION_UUID)).isTrue();
+    assertThat(loadedTemplateExists(QUALITY_PROFILE_TYPE.toLowerCase(), key1)).isTrue();
+    assertThat(loadedTemplateExists("other type", key2)).isTrue();
+    assertThat(loadedTemplateExists(QUALITY_PROFILE_TYPE + '.' + md5Hex(key2.getBytes(UTF_8)), DEFAULT_ORGANIZATION_UUID)).isTrue();
+    assertThat(db.countRowsOfTable(TABLE_LOADED_TEMPLATES)).isEqualTo(4);
+  }
+
+  @Test
+  public void execute_is_reentrant() throws Exception {
+    setupDefaultOrganization();
+    String key = "blabla";
+    insertLoadedTemplate(QUALITY_PROFILE_TYPE, key);
+
+    underTest.execute();
+
+    underTest.execute();
+
+    assertThat(loadedTemplateExists(QUALITY_PROFILE_TYPE + '.' + md5Hex(key.getBytes(UTF_8)), DEFAULT_ORGANIZATION_UUID)).isTrue();
+    assertThat(db.countRowsOfTable(TABLE_LOADED_TEMPLATES)).isEqualTo(1);
+  }
+
+  private void setupDefaultOrganization() {
+    insertDefaultOrganizationUuid(DEFAULT_ORGANIZATION_UUID);
+    insertOrganization(DEFAULT_ORGANIZATION_UUID);
+  }
+
+  private void insertOrganization(String uuid) {
+    db.executeInsert(
+      TABLE_ORGANIZATIONS,
+      "UUID", uuid,
+      "KEE", uuid,
+      "NAME", uuid,
+      "GUARDED", String.valueOf(false),
+      "CREATED_AT", "1000",
+      "UPDATED_AT", "1000");
+  }
+
+  private void insertDefaultOrganizationUuid(String defaultOrganizationUuid) {
+    db.executeInsert(
+      "INTERNAL_PROPERTIES",
+      "KEE", "organization.default",
+      "IS_EMPTY", "false",
+      "TEXT_VALUE", defaultOrganizationUuid);
+  }
+
+  private void insertLoadedTemplate(String type, String key) {
+    db.executeInsert(
+      TABLE_LOADED_TEMPLATES,
+      "TEMPLATE_TYPE", type,
+      "KEE", key);
+  }
+
+  private boolean loadedTemplateExists(String type, String key) {
+    return !db.selectFirst(String.format("select id from loaded_templates where template_type='%s' and kee='%s'", type, key)).isEmpty();
+  }
+
+  @Test
+  public void name() {
```

Comment:
@simonbrandhof your review missed that one piece of temporary code that I had forgotten to remove ;)